** WIP ** Voting UI

nickfujita opened the pull request:
This is a draft of the voting UI based on wireframes from @comountainclimber 

Will look to modularize and clean up more once design and flow gets tied down.

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+import classNames from 'classnames'
+
+type Props = {
+  hideModal: Function,
```
`hideModal: () => any`

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+  votes: Object
+}
+
+const textStrings = {
```
I partially understand why you chose to do this, but I think its much better if you put those strings in JSX elements. We sure can use `i18n` in v2, but this isn't needed here.

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+  VOTING_EXPLANATION: 'Each NEO node can vote for the candidtates. The number of NEON in the current voting account will be automatically calculated as the number of the candidate\'s votes. When voting for multiple candidates, each candidate gets the votes equal to the NEO number of the current voting account. For example, if there are 100 NEO in the current account and the three candidates are v ...
+  PUBLIC_ADDRESS: 'Public address of candidate',
+  CONFIRMATION_TITLE: 'Confirm votes',
+  CONFIRMATION_MESSAGE: 'You will be casting <number_of_votes> votes, for each of the following candidates:',
```
Please see my comment below how to handle this ^

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+          }
+        }}
+      >
+        {textStrings.CONFIRMATION_MESSAGE.replace('<number_of_votes>', votesAvailable.toString())}
```
Please use lodash `template` to achieve safe string replacement 

```
import { template } from 'lodash'

template(
  'You will be casting ${numberOfVotes} votes, for each of the following candidates:'
)({ numberOfVotes: votesAvailable.toString() })
```

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+  hasVotes: Function
+
+  componentWillMount () {
+    this.showInfoModal = this.showInfoModal.bind(this)
```
You wrote these so you could better test it? Because you could just use es6 arrow functions, and you won't need to bind.

Also, if you do happen to want to use this, then place it in the constructor

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+          {Object.keys(votes).reduce((elements, key, index) => {
+            if (votes[key]) {
+              elements.push(
+                <li key={index}>{placehoderData[Number(key)].address}</li>
```
Could you use a more unique key? Its best when its actual ids, but if you use index then - perhaps `voteRow${index}`

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+    return placehoderData.map((data, index) => {
+      const {address, votes} = data
+      return (
+        <div key={index} className={styles.row}>
```
Could you use a more unique key? Its best when its actual ids, but if you use index then - perhaps `nodeRow${index}`

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+      >
+        {textStrings.CONFIRMATION_MESSAGE.replace('<number_of_votes>', votesAvailable.toString())}
+        <ul className={styles.addressList}>
+          {Object.keys(votes).reduce((elements, key, index) => {
```
could you put types on `elements, key, index`

evgenyboxer commented on app/components/Modals/VotingModal/VotingModal.jsx:
```diff
@@ -0,0 +1,246 @@
+        hideModal={hideModal}
+        style={{
+          content: {
+            width: '925px',
```
is there a reason why you used pixels here and `rem` in the other usages of modal?

nickfujita commented:
@comountainclimber thanks for helping out with the updates to address the feedback from @evgenyboxer  :)

I moved forward on updating the UI to:
- remove the double modals
- contain short info on screen about voting with link to more info
- Added the votes already cast from this wallet, next to the votes cast overall for each node
- moved the vote confirmation to the same screen where the voting selection takes place
- fixed a few remaining linting concerns

drptbl commented:
After clicking on "More info" hyperlink, we're redirected to the start page of neon wallet.

**Environment:**
This is Windows-only issue, it doesn't occur on macOS.

**Reproduction steps:**
1. Start Neon Wallet
2. Select any saved wallet
3. Enter your passphrase
4. Click on "Login" button
5. Click on "Vote for consensus node" button
6. Click on "More info" hyperlink

**Video:**
![votefor](https://user-images.githubusercontent.com/8177587/42423846-8b66df60-8301-11e8-8a37-f6d2fb18eaba.gif)
